### PR TITLE
[gym][fastlane_core] add platform support for visionOS

### PR DESCRIPTION
**Pull request by PinkidG**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

With the upcoming release of the Vision Pro, we'll have a new platform that gym needs to support. While the device uses the same architecture as iOS devices, it does have its own platforms (`xros`, `xrsimulator`)

### Description

Added a new platform identifier and cases to `project.rb` make sure that new cases are considered in gym.

### CODE REVIEW

**Comment by triplef** [comment]
Thanks for this PR @PinkidG! We’d love to see this supported as well. ✨ 

Any chance for this to get merged @joshdholtz? 🙏

File: gym/spec/platform_detection_spec.rb [hunk 1]
```diff
@@ -24,4 +24,22 @@
     expect(Gym.building_for_mac?).to eq(true)
     expect(Gym.building_multiplatform_for_mac?).to eq(true)
   end
+
+  it "detects the correct platform for a visionOS project", requires_xcodebuild: true do
+    options = { project: "./gym/examples/visionos/VisionExample.xcodeproj", sdk: "xros", skip_package_dependencies_resolution: true }
+    Gym.config = FastlaneCore::Configuration.create(Gym::Options.available_options, options)
```
Comment (lacostej) [inline gym/spec/platform_detection_spec.rb:30]: maybe we could write

```
  it "...", requires_xcodebuild: true, if: FastlaneCore::Helper.xcode_at_least?('15.0') do
```

and get rid of the if/else.
?

File: gym/examples/visionos/Packages/RealityKitContent/Package.realitycomposerpro/ProjectData/main.json
Comment (lacostej) [inline gym/examples/visionos/Packages/RealityKitContent/Package.realitycomposerpro/ProjectData/main.json:1]: not familiar with visionos. Where does this example code come from?

**Comment by triplef** [comment]
@PinkidG any chance you could incorporate the feedback from @lacostej? Would be amazing to see this merged soon with Apple seemingly getting ready for launch of the platform. 🙏

File: gym/spec/platform_detection_spec.rb (hunk 1 above)
Comment (PinkidG) [reply gym/spec/platform_detection_spec.rb:30]: Sure, thanks!

File: gym/examples/visionos/Packages/RealityKitContent/Package.realitycomposerpro/ProjectData/main.json
Comment (PinkidG) [reply gym/examples/visionos/Packages/RealityKitContent/Package.realitycomposerpro/ProjectData/main.json:1]: @lacostej That is the default project that is generated from Xcode when starting a new project. I didn't really add anything to the project, I hope that's alright 🙂.

**Comment by triplef** [comment]
@lacostej is there anything else that needs to be done here in order to land this in a release? 🚀

**Comment by PinkidG** [comment]
Our teams dev teams would love to have proper visionOS support in fastlane. 

@lacostej or @rogerluan is there anything else required to get this merged?

Thanks!

File: gym/lib/gym/module.rb [hunk 2]
```diff
@@ -33,8 +33,8 @@ def building_for_ios?
         # Can be iOS project and build for mac if catalyst
         return false if building_mac_catalyst_for_mac?
 
-        # Can be iOS project if iOS, tvOS, or watchOS
-        return Gym.project.ios? || Gym.project.tvos? || Gym.project.watchos?
+        # Can be iOS project if iOS, tvOS, watchOS or visionOS
```
Comment (rogerluan) [inline gym/lib/gym/module.rb:36]: Nit:

```suggestion
        # Can be iOS project if iOS, tvOS, watchOS, or visionOS
```

File: gym/examples/visionos/Packages/RealityKitContent/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# RealityKitContent
+
+A description of this package.
```
Comment (rogerluan) [inline gym/examples/visionos/Packages/RealityKitContent/README.md:3]: Maybe add something here or delete this file?

```suggestion
A visionOS demo project used to run integration test in `gym/spec/platform_detection_spec.rb` (and maybe in the future in other tests too).
```

**Review by rogerluan — APPROVED** [review]
Thanks for this contribution @PinkidG ! 🙌  I really appreciate you adding the demo project for the integration test, this is really helpful. I haven't done any work with visionOS yet so I don't understand some of those 3D model and scene files that you added but I'm sure it's ok for integration testing purposes :)

Looking forward to seeing great visionOS apps being built! 🚀

**Comment by triplef** [comment]
Since the question [came up](https://github.com/fastlane/fastlane/discussions/21802) in discussions: does this PR also unlock visionOS support for _deliver_ (e.g. uploading visionOS binaries to the App Store)?

**Comment by rogerluan** [comment]
Replied in the discussion @triplef 🙇

**Comment by alpennec** [comment]
Hey @rogerluan,

Can you please see my comment here: https://github.com/fastlane/fastlane/pull/21841#issuecomment-2186501421
I'm not able to use `xros` as a value for the `platform` parameter using fastlane 2.221.1.

`Error setting value 'xros' for option 'platform'`
`[!] The platform can only be ios, appletvos, or osx`

Is there anything specific to do? Thanks!

Thanks,
Axel

**Pull request merged**